iroha/src/torii/mod.rs

```
@@ -208,41 +201,29 @@ impl<Q: QueueTrait, S: SumeragiTrait, W: WorldTrait> Torii<Q, S, W> {
             .or(ws_router)
             .with(warp::trace::request());
 
-        #[allow(clippy::unwrap_used)]
-        let addr = self
-            .config
-            .torii_api_url
-            .to_socket_addrs()
-            .wrap_err("Failed to get socket addr")?
-            .next()
-            .unwrap();
-
-        let (handle_requests_result, _) = tokio::join!(
-            Network::listen(
-                Arc::clone(&state),
-                &self.config.torii_p2p_url,
-                handle_requests
-            ),
-            warp::serve(router).run(addr),
-        );
-        handle_requests_result
+        #[allow(clippy::expect_used)]
+        if let Ok(mut i) = self.config.torii_api_url.to_socket_addrs() {
+            let addr = i.next().expect("Failed to get socket addr");
+            warp::serve(router).run(addr).await;
+        } else {
+            iroha_logger::error!("Failed to get socket addr");
+        }
```

I guess you can return this as errors

---

Done.